### PR TITLE
PP-6271 Allow user to update Cookie Analytic preferences via the /coo…

### PR DESCRIPTION
**Pull request by iqbalgds**
…kie page

- Added a form at the bottom of the /Cookie page
  - Allows user to consent or decline the use of Analytic cookies.
- When they go the page, if they have not set any cookie prefs previously, then it shows
 warning saying they can update their prefs.
- If they click YES or NO, a cookie is saved which remembers their preference.
- All functionality in cookie-settings.js file.
- Requires the work done for the Cookie Banner.
- Added SCSS styling.
- Added Unit Tests

### CODE REVIEW

File: source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    document.querySelector(
+      ".pay-cookie-settings-alert--success"
+    ).style.display = "none";
+  });
+
+  describe("initialising the component", () => {
+    it("hides the cookie banner if present", () => {
+      window.GovUkPay.CookieSettings.checkForFormAndInit();
+
+      expect(
```
Comment (kbottla) [inline source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js:55]: May be have excepts in one line ? Linter could move to one line or we could check if team has convention

File: source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+        expect(
+          document.querySelector(".pay-cookie-settings-alert--warning").style
+            .display
+        ).toEqual("none");
+      });
+    });
+
+    describe("initialising the Cookie Settings radio buttons", () => {
+      beforeEach(() => {
```
Comment (kbottla) [inline source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js:84]: Can remove `beforeEach` from this block

File: source/pay-product-page/stylesheets/modules/_cookie-settings.scss [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    margin: govuk-spacing(3) 0 govuk-spacing(6) 0;
+    text-align: left;
+    position: relative;
+    clear: both;
+    border-style: solid;
+    border-width: 5px;
+}
+
+.pay-cookie-settings-alert--warning {
+    // display: none;
```
Comment (kbottla) [inline source/pay-product-page/stylesheets/modules/_cookie-settings.scss:20]: Can remove this commented line

File: source/cookies.html.erb [hunk 4]
```diff
@@ -18,6 +18,7 @@ title: Cookies
   <main id="main-content" class="govuk-main-wrapper" role="main">
     <div class="govuk-grid-row">
       <div class="govuk-grid-column-two-thirds">
+      <%= partial 'partials/cookie-settings-confirmation.erb' %>
```
[the remark this answers is not in the document]
Comment (iqbalgds) [reply source/cookies.html.erb:21]: Done

File: source/cookies.html.erb [hunk 5]
```diff
@@ -105,6 +106,74 @@ title: Cookies
           </tr>
         </tbody>
       </table>
+
+      <h2 class="govuk-heading-m">Analytics cookies (optional)</h2>
```
[the remark this answers is not in the document]
Comment (iqbalgds) [reply source/cookies.html.erb:110]: Done

File: source/partials/_cookie-settings-confirmation.erb [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    <h2>
+        There was a problem saving your settings
+    </h2>
+    
+    <div class="pay-cookie-settings-alert__body">
+        Please select 'Yes' or 'No'.
+    </div>    
+</div>
+
+<div data-module="pay-cookie-settings-alert" class="pay-cookie-settings-alert pay-cookie-settings-alert--warning" id="pay-cookie-settings-warning" tabindex="0">
```
[the remark this answers is not in the document]
Comment (iqbalgds) [reply source/partials/_cookie-settings-confirmation.erb:11]: Done - updated SASS styling.

File: source/pay-product-page/javascripts/cookie-settings/cookie-settings.js [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+            warningMessage.style.display = "none";
+        }
+    };
+
+    var hideCookieBanner = function() {
+        var cookieBanner = document.querySelector(".pay-cookie-banner");
+        if (cookieBanner !== null) {
+            cookieBanner.style.display = "none";
+        }
+    };
```
[the remark this answers is not in the document]
Comment (iqbalgds) [reply source/pay-product-page/javascripts/cookie-settings/cookie-settings.js]: Done

File: source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+    window.GovUkPay = window.GovUkPay || {};
+    window.GovUkPay.InitAnalytics = {};
+    window.GovUkPay.InitAnalytics.InitialiseAnalytics = jest.fn();
+
+    require("../helpers/cookie/cookie-functions");
+    require("./cookie-settings");
+
+    JsCookie.remove("govuk_pay_cookie_policy");
+
+    document.body.innerHTML = testCookiePageTemplate;
```
[the remark this answers is not in the document]
Comment (iqbalgds) [reply source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js:42]: Will raise a separate ticket for this, will need to investigate

File: source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js (hunk 1 above)
Comment (iqbalgds) [reply source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js:55]: Will raise separate PR with running the `standard` linter

File: source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js (hunk 2 above)
Comment (iqbalgds) [reply source/pay-product-page/javascripts/cookie-settings/cookie-settings.test.js:84]: Done

File: source/pay-product-page/stylesheets/modules/_cookie-settings.scss (hunk 3 above)
Comment (iqbalgds) [reply source/pay-product-page/stylesheets/modules/_cookie-settings.scss]: Cleaned up this file - this is no longer here.

File: source/pay-product-page/javascripts/cookie-banner/cookie-banner.test.js [hunk 9]
```diff
@@ -106,18 +106,18 @@ describe("Cookie Banner", () => {
     });
   });
 
-  describe('Confirmation mesage', () => {
+  describe("Confirmation mesage", () => {
```
[the remark this answers is not in the document]
Comment (iqbalgds) [reply source/pay-product-page/javascripts/cookie-banner/cookie-banner.test.js:109]: Will do as a separate PR

**Pull request merged**